### PR TITLE
Training Models using Built-in Keras Function

### PR DESCRIPTION
**Pull request by usimarit**
- [x] Implement Slice Dataset that yield data in keras format `(x, y)`
- [x] Implement TFRecord Dataset that yield data in keras format `(x, y)`
- [x] Add new `RnntLoss` subclass of `tf.keras.losses.Loss` to use in `compile` function of `tf.keras.Model` 
- [x] Add Transducer with custom `train_step` and `test_step` that uses `RnntLoss`
- [x] Add new `CtcLoss` subclass of `tf.keras.losses.Loss` to use in `compile` function of `tf.keras.Model`
- [x] Add CtcModel with custom `train_step` and `test_step` that uses `CtcLoss`

### CODE REVIEW

File: tensorflow_asr/models/keras/transducer.py [hunk 1]
```diff
@@ -47,14 +46,17 @@ def call(self, inputs, training=False, **kwargs):
...
             y_pred = self(x, training=True)
-            loss = self.loss(y, y_pred)
+            loss = self.loss(y_true, y_pred)
         gradients = tape.gradient(loss, self.trainable_variables)
         self.optimizer.apply_gradients(zip(gradients, self.trainable_variables))
+        return {"loss": loss}
 
+    def test_step(self, batch):
+        x, y_true = batch
+        y_pred = self(x, training=True)
```
Comment (dathudeptrai) [inline tensorflow_asr/models/keras/transducer.py:60]: shoud be `y_pred = self(x, training=False)` ? 
`

File: tensorflow_asr/models/keras/transducer.py [hunk 2]
```diff
@@ -47,14 +46,17 @@ def call(self, inputs, training=False, **kwargs):
...
-    def train_step(self, data):
-        data = data_adapter.expand_1d(data)
-        x, y, sample_weight = data_adapter.unpack_x_y_sample_weight(data)
-
+    def train_step(self, batch):
+        x, y_true = batch
         with tf.GradientTape() as tape:
             y_pred = self(x, training=True)
-            loss = self.loss(y, y_pred)
+            loss = self.loss(y_true, y_pred)
```
Comment (dathudeptrai) [inline tensorflow_asr/models/keras/transducer.py:53]: Pls adding Mixed precision support here

File: tensorflow_asr/models/keras/transducer.py (hunk 1 above)
Comment (usimarit) [reply tensorflow_asr/models/keras/transducer.py]: fixed

File: tensorflow_asr/models/keras/transducer.py (hunk 2 above)
Comment (usimarit) [reply tensorflow_asr/models/keras/transducer.py:66]: loss scaling added, in fact, it doesn't need loss scaling because the loss is already too big at the beginning but I added anyway
